packages/application-components/src/components/main-pages/form-main-page/form-main-page.tsx

```diff
@@ -0,0 +1,120 @@
+import type { ReactElement, ReactNode, SyntheticEvent } from 'react';
+import { sharedMessages } from '@commercetools-frontend/i18n';
+import CustomFormMainPage from '../custom-form-main-page';
+import PageHeaderTitle from '../../internals/page-header-title';
+
+// NOTE: the `MessageDescriptor` type is exposed by `react-intl`.
+// However, we need to explicitly define this otherwise the prop-types babel plugin
+// does not recognize the object shape.
+type MessageDescriptor = {
+  id: string;
+  description?: string | object;
+  defaultMessage?: string;
+};
+type Label = string | MessageDescriptor;
+
+type FormMainPageProps = {
+  /**
+   * The title of the page.
+   */
+  title?: string;
+  /**
+   * The subtitle of the page.
+   */
+  subtitle?: string | ReactElement;
+  /**
+   * Replaces the title/subtitle row with a custom one (for special use cases)
+   */
+  customTitleRow?: ReactNode;
+  /**
+   * Any React node displayed as the content within the page.
+   */
+  children: ReactNode;
+  /**
+   * Determines if the form controls should be rendered.
+   */
+  hideControls?: boolean;
+  /**
+   * Indicates whether the primary button is deactivated or not.
+   */
+  isPrimaryButtonDisabled?: boolean;
+  /**
+   * Indicates whether the secondary button is deactivated or not.
+   */
+  isSecondaryButtonDisabled?: boolean;
+  /**
+   * Use this prop to pass `data-` attributes to the primary button
+   */
+  dataAttributesPrimaryButton?: { [key: string]: string };
+  /**
+   * Use this prop to pass `data-` attributes to the secondary button.
+   */
+  dataAttributesSecondaryButton?: { [key: string]: string };
+  /**
+   * `Save` | The label for the primary button as a string, or as an Intl message.
+   */
+  labelPrimaryButton?: Label;
+  /**
+   * `Cancel` | The label for the secondary button as a string, or as an Intl message.
+   */
+  labelSecondaryButton?: Label;
+  /**
+   * The Icon for the secondary button label
+   */
+  labelSecondaryButtonIcon?: ReactElement;
```

Comment:
Is this a new prop? Any particular reason for having it?

In any case, for consistency we should probably add this prop to the other similar page components.

Reply:
The new prop `labelSecondaryButtonIcon` helps display an icon for the secondary button. 
This was necessary because of the new design. 

The new Prop will be added to the other form pages.

Reply:
Updated [bbfa237](https://github.com/commercetools/merchant-center-application-kit/pull/2745/commits/bbfa237abed160a03993c922f4e449f9bee9d081)